parts/k8s/kubernetesinstalls.sh

```diff
@@ -46,7 +49,7 @@ installGPUDrivers() {
     wait_for_apt_locks
     retrycmd_if_failure 120 5 25 apt-key add /tmp/aptnvidia.gpg || exit $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
     wait_for_apt_locks
-    retrycmd_if_failure_no_stats 120 5 25 curl -fsSL https://nvidia.github.io/nvidia-docker/ubuntu16.04/amd64/nvidia-docker.list > /tmp/nvidia-docker.list || exit  $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
+    retrycmd_if_failure_no_stats 120 5 25 curl -fsSL https://nvidia.github.io/nvidia-docker/ubuntu${UBUNTU_RELEASE}/nvidia-docker.list > /tmp/nvidia-docker.list || exit  $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
```

Comment:
I think we're missing a `/amd64` substring here

Reply:
that was on purpose. `https://nvidia.github.io/nvidia-docker/ubuntu18.04/amd64/nvidia-docker.list` doesn't exist but `https://nvidia.github.io/nvidia-docker/ubuntu18.04/nvidia-docker.list` does. `https://nvidia.github.io/nvidia-docker/ubuntu16.04/nvidia-docker.list` and `https://nvidia.github.io/nvidia-docker/ubuntu16.04/amd64/nvidia-docker.list` are equivalent.
We don't support GPU + 18.04 because 18.04 is only supported with Moby so GPU is only supported with 16.04 + docker-engine. I could revert this change and leave 16.04 hardcoded, or I can leave it so the code adapts if we do support GPU with 18.04 in the future.

Reply:
I'll run an vanilla ubuntu 16.04 + GPU test to confirm this

Reply:
Thanks for the explanation. Does docker-engine literally not install on 18.04?

Reply:
The 18.04 distribution isn't available in the docker-engine repo https://apt.dockerproject.org/repo/dists/

Reply:
We should probably just add an additional "is 16.04" guard to all the `installGPUDrivers` and `ensureGPUDrivers` references, and should not enable 18.04 + N series SKUs at api model validation time.

Reply:
https://github.com/Azure/aks-engine/pull/223/files#diff-f45af0b14c127459636440a5981a6934R782 :)

Reply:
How about packer script and CSE?

Reply:
packer only installs docker-engine and GPU drivers if `*"docker-engine"*` flag is explicitly specified. Ubuntu 18.04 is also an explicit flag. We shouldn't build VHD with 18.04 + docker-engine both explicitly enabled. If we do, it will fail and we'll realize it's because there is no 18.04 docker-engine release.

I don't think we should add additional guards to install functions in CSE. Validation should happen at validation time, not later when we are installing things. Maybe in the near future we'll allow GPU installs with 18.04, but for now we block that configuration at apimodel validation level. Same reason we guard the `installGPUDrivers` function in CSE with "isDockerEngine", that's an apimodel configuration level validation. Otherwise we'd have to check every allowed config in CSE which would result in a bunch of nested if/else statements. Blocking apimodels with 18.04 and N-series nodes is enough IMO.

Reply:
verified that vanilla ubuntu 16.04 distro + GPU on this branch deploys correctly.